app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class SessionsController < ApplicationController
+  before_action :load_user_by_email, only: :create
+  before_action :check_authentication, only: :create
+
+  def new; end
+
+  def create
+    log_in @user
+    redirect_to @user
+  end
+
+  def destroy
+    log_out
+    redirect_to root_path
+  end
+
+  def load_user_by_email
```

Comment:
```suggestion

  private
  def load_user_by_email
```
Thêm private trước các method k phải action controller này nhé